### PR TITLE
Send users directly to web when we do not support Native GPay 

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T373410

The idea is to use the current logic in the `GooglePayViewModel` that decides to show in the Donation dialog or CustomTab.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/main/MainFragment.kt [hunk 1]
```diff
@@ -465,7 +476,26 @@ class MainFragment : Fragment(), BackPressedHandler, MenuProvider, FeedFragment.
...
-        (requireActivity() as? BaseActivity)?.launchDonateDialog()
+        if (showDonationBottomSheet) {
+            (requireActivity() as? BaseActivity)?.launchDonateDialog()
+        } else {
+            DonateDialog.launchDonateLink(requireContext())
+        }
+    }
+
+    private fun checkDonateConfig() {
+        CoroutineScope(Dispatchers.IO).launch {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/main/MainFragment.kt:487]: This seems like a lot of highly specific business logic being placed into `MainFragment`. Can we put this into a ViewModel, or into `DonationConfigHelper`?

File: app/src/extra/java/org/wikipedia/donate/GooglePayViewModel.kt [hunk 2]
```diff
@@ -85,24 +84,7 @@ class GooglePayViewModel : ViewModel() {
...
-                }
-                paymentMethodsCall.await().response?.let { response ->
-                    Prefs.paymentMethodsLastQueryTime = now
-                    response.paymentMethods.find { it.type == GooglePayComponent.PAYMENT_METHOD_NAME }?.let {
-                        Prefs.paymentMethodsMerchantId = it.configuration?.merchantId.orEmpty()
-                        Prefs.paymentMethodsGatewayId = it.configuration?.gatewayMerchantId.orEmpty()
-                    }
-                }
-            }
+            async { updatePaymentMethodsPreferences() }.await()
```
Comment (dbrant) [inline app/src/extra/java/org/wikipedia/donate/GooglePayViewModel.kt:87]: This doesn't need to be inside `async{}.await()`

**Review by dbrant — CHANGES_REQUESTED** [review]
Couldn't we just put all this logic in `DonateViewModel`? It's already doing most of the work for us.  (i.e. in addition to checking `isGooglePayAvailable()`, you can also check if the country code is allowed.)

File: app/src/extra/java/org/wikipedia/donate/GooglePayViewModel.kt (hunk 2 above)
Comment (cooltey) [reply app/src/extra/java/org/wikipedia/donate/GooglePayViewModel.kt:87]: Sure!

File: app/src/main/java/org/wikipedia/main/MainFragment.kt (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/main/MainFragment.kt]: You are right. Moved to `DonateViewModel`.

**Comment by cooltey** [comment]
> Couldn't we just put all this logic in `DonateViewModel`? It's already doing most of the work for us. (i.e. in addition to checking `isGooglePayAvailable()`, you can also check if the country code is allowed.)

I moved the logic to `DonateViewModel`, it works great but the user will see the progress bar in the bottom sheet and then automatically redirect to the donation page. 🤔

**Comment by cooltey** [comment]
Thanks @dbrant

**Pull request merged**